File: backend/internal/domain/product.go
package domain

import (
	"time"
)

type Product struct {
	ID          uint      `json:"id" gorm:"primaryKey"`
	Name        string    `json:"name" gorm:"not null"`
	Description string    `json:"description"`
	Price       float64   `json:"price" gorm:"not null"`
	Stock       int       `json:"stock" gorm:"not null;default:0"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ProductRepository interface {
	Create(product *Product) error
	GetByID(id uint) (*Product, error)
	GetAll() ([]*Product, error)
	Update(product *Product) error
	Delete(id uint) error
	UpdateStock(id uint, stock int) error
	UpdateStockWithTx(tx Transaction, id uint, stock int) error
}

File: app_pos/app/src/main/java/com/example/app_pos/domain/entity/Product.kt
package com.example.app_pos.domain.entity

data class Product(
    val id: String,
    val name: String,
    val description: String,
    val price: Double,
    val stockQuantity: Int,
    val category: String,
    val imageUrl: String? = null,
    val isActive: Boolean = true,
    val createdAt: String,
    val updatedAt: String
)